02-peer-protocol.md

```diff
@@ -1033,6 +1036,21 @@ A sending node:
   - MUST increase the value of `id` by 1 for each successive offer.
   - if it is relaying a payment inside a blinded route:
     - MUST set `blinding_point` (see [Route Blinding](04-onion-routing.md#route-blinding))
+  - if it is the original source of the HTLC:
+    - if it does not expect immediate fulfillment upon receipt by the 
+      final destination: 
+      - SHOULD set `endorsed` to `0`.
+    - otherwise:  
+      - SHOULD set `endorsed` to `1`. 
+      - MAY choose to set `endorsed` to `0` to mimic endorsement patterns of 
+        HTLCs it has forwarded.
+  - otherwise: 
+    - if `endorsed` is present and non-zero for the corresponding incoming HTLC
+      AND the incoming peer is considered to have sufficient local reputation
+      (see [Local Reputation](recommendations/local-resource-conservation.md#local-reputation)): 
+      - SHOULD set `endorsed` to `1`
+    - otherwise: 
+      - SHOULD set `endorsed` to `0`.
```

Comment:
Is there a reason to prefer including the tlv with a non-endorsement as opposed to omitting it? As far as I can tell, the downstream node interprets a missing endorsement tlv exactly the same as setting it to 0.  I suppose I'm wondering if there's any reason to track upstream endorsement as a ternary option.  Does it help peers to signal that the node is actively resource bucketing / tracking reputation with endorsements even if the payments carry no endorsement?

Reply:
iirc the reason what that we want to know the people who are using the `endorsed` and the people that does not.

I also remember that we were discussing a feature bit that makes the `endorsed` optional when it is `0`

Reply:
Mh @carlaKC I think that the blips were overriding this, what is the correlation between the two proposals?